### PR TITLE
pre-transfer-leader: use log term instead of term

### PR DESCRIPTION
**Pull request by BusyJay**
###  What have you changed?

raft-rs forbids setting term for MsgTransferLeader, so use log term
instead.

###  What is the type of the changes?
- Bugfix

###  How is the PR tested?
- Manual test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No.

###  Does this PR affect `tidb-ansible`?
No.

### CODE REVIEW

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -2263,7 +2268,7 @@ impl Peer {
         msg.set_to(self.leader_id());
         msg.set_msg_type(eraftpb::MessageType::MsgTransferLeader);
         msg.set_index(self.get_store().applied_index());
-        msg.set_term(self.term());
+        msg.set_log_term(self.term());
```
Comment (hicqu) [inline src/raftstore/store/peer.rs]: For followers `log term` may be not equal to `term`. Is it ok?

Comment (BusyJay) [inline src/raftstore/store/peer.rs]: It's checked at L2227.

Comment (BusyJay) [reply src/raftstore/store/peer.rs]: It's checked at L2227.

**Pull request merged**